radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixJsonRpcClient.java

```diff
@@ -77,19 +73,19 @@ public RadixJsonRpcClient(WebSocketClient wsClient) {
 			.publish()
 			.refCount();
 
-		if (!CHECK_API_VERSION) {
-			this.serverApiVersion = Single.just(API_VERSION);
-		} else {
-			this.serverApiVersion = jsonRpcCall("Api.getVersion")
-				.map(result -> result.getAsJsonObject().get("version").getAsInt())
-				.cache();
-		}
+		this.serverApiVersion = SingleSubject.create();
+		this.serverApiVersion.onSuccess(API_VERSION);
+		jsonRpcCall("Api.getVersion")
+			.map(result -> result.getAsJsonObject().get("version").getAsInt())
+			.subscribe(serverApiVersion::onSuccess, e -> LOGGER.error("Error while requesting Api.getVersion: " + e, e));
 
 		Serialization serialization = Serialize.getInstance();
-		this.universeConfig = jsonRpcCall("Universe.getUniverse")
+
+		this.universeConfig = SingleSubject.create();
+		jsonRpcCall("Universe.getUniverse")
 			.map(element -> GsonJson.getInstance().stringFromGson(element))
 			.map(result -> serialization.fromJson(result, RadixUniverseConfig.class))
-			.cache();
+			.subscribe(universeConfig::onSuccess);
```

Comment:
I don't think we want JsonRpcClient making the subscribe calls. These should be called by an outside class. It feels wrong that instantiating a class automatically makes network calls.

Reply:
Agreed, fixed.